src/components/Capture/withCrossDeviceWhenNoCamera.js

```diff
@@ -34,7 +35,7 @@ export default WrappedComponent =>
         }
         const step = 0
         const excludeStepFromHistory = true
-        changeFlowTo('crossDeviceSteps', step, excludeStepFromHistory)
+        isDesktop && changeFlowTo('crossDeviceSteps', step, excludeStepFromHistory)
```

Comment:
opinion, this block:

```
 if (cameraRequiredButNoneDetected || forceCrossDevice) {
   ....
```

currently has two guards:

``` 
if (this.props.mobileFlow) { .... return }  
if (this.props.mobileFlow && !this.props.uploadFallback) { .... return } 
```

and what you adding is another guard:

```
isDesktop && sideEffect()
```

Why not maintain the guard logic:

``` 
if (this.props.mobileFlow) { .... return }  
if (this.props.mobileFlow && !this.props.uploadFallback) { .... return } 
if( isDesktop) { .... return }

sideEffect
```

Reading on guards: https://medium.com/@scadge/if-statements-design-guard-clauses-might-be-all-you-need-67219a1a981a

What do you think? Does it improve readability?

Reply:
It might. I will keep it the same guard for consistency. Thanks for the article link 🙏